### ISSUE TITLE
Markdown output polluted with previous style setting

### ISSUE THREAD

**Issue by bmihaila-bd** [jazzband/prettytable#341]
### What did you do?
Emit a table in markdown format
### What did you expect to happen?
The markdown output should be parsable and rendering as a table in markdown viewers (e.g. Github)
### What actually happened?
The markdown output is not rendered at all as a table
### What versions are you using?

* PrettyTable: 
3.11.0

### Explanation
It seems that previous styles are not cleaned up when setting the MARKDOWN style, so they mess up the markdown output. 
```python
# code goes here
table.set_style(prettytable.SINGLE_BORDER)
...
table.set_style(prettytable.MARKDOWN)
# results in below output, with corners and some line connectors coming from the BORDER style. That breaks the markdown parsers.
┌ :───────────────────: ┐
| Namespace    |  Count |
├ :────────────|──────: ┤
| bla      |     24 |
| blu |      1 |
| blam       |      1 |
```

The simple fix seems to be to call `table._set_default_style()` in the `def _set_markdown_style(self)` method. That results in this markdown parsable output:
```
| Namespace    | Count |
| :------------|-----: |
| bla       |    24 |
| blu |     1 |
| blam       |     1 |
```

Below I can show how both above look with  the Github parser here:
broken:
┌ :───────────────────: ┐
| Namespace    |  Count |
├ :────────────|──────: ┤
| bla      |     24 |
| blu |      1 |
| blam       |      1 |

fixed:
| Namespace    | Count |
| :------------|-----: |
| bla       |    24 |
| blu |     1 |
| blam       |     1 |

**Comment by hugovk** [comment]
# Repro 

Here's self-contained script to reproduce:

```python
from prettytable import TableStyle, PrettyTable

table = PrettyTable()
table.field_names = ["City name", "Area", "Population", "Annual Rainfall"]
table.add_row(["Adelaide", 1295, 1158259, 600.5])
table.add_row(["Brisbane", 5905, 1857594, 1146.4])
table.add_row(["Darwin", 112, 120900, 1714.7])
table.add_row(["Hobart", 1357, 205556, 619.5])
table.add_row(["Sydney", 2058, 4336374, 1214.8])
table.add_row(["Melbourne", 1566, 3806092, 646.9])
table.add_row(["Perth", 5386, 1554769, 869.4])

table.set_style(TableStyle.SINGLE_BORDER)
table.set_style(TableStyle.MARKDOWN)

print(table)
```

Outputs:

```md
| City name | Area | Population | Annual Rainfall |
├ :───────: | :──: | :────────: | :─────────────: ┤
|  Adelaide | 1295 |  1158259   |      600.5      |
|  Brisbane | 5905 |  1857594   |      1146.4     |
|   Darwin  | 112  |   120900   |      1714.7     |
|   Hobart  | 1357 |   205556   |      619.5      |
|   Sydney  | 2058 |  4336374   |      1214.8     |
| Melbourne | 1566 |  3806092   |      646.9      |
|   Perth   | 5386 |  1554769   |      869.4      |
```

But doesn't render properly:

| City name | Area | Population | Annual Rainfall |
├ :───────: | :──: | :────────: | :─────────────: ┤
|  Adelaide | 1295 |  1158259   |      600.5      |
|  Brisbane | 5905 |  1857594   |      1146.4     |
|   Darwin  | 112  |   120900   |      1714.7     |
|   Hobart  | 1357 |   205556   |      619.5      |
|   Sydney  | 2058 |  4336374   |      1214.8     |
| Melbourne | 1566 |  3806092   |      646.9      |
|   Perth   | 5386 |  1554769   |      869.4      |


# Workaround

Set the default style in between:

```python
from prettytable import TableStyle, PrettyTable

table = PrettyTable()
table.field_names = ["City name", "Area", "Population", "Annual Rainfall"]
table.add_row(["Adelaide", 1295, 1158259, 600.5])
table.add_row(["Brisbane", 5905, 1857594, 1146.4])
table.add_row(["Darwin", 112, 120900, 1714.7])
table.add_row(["Hobart", 1357, 205556, 619.5])
table.add_row(["Sydney", 2058, 4336374, 1214.8])
table.add_row(["Melbourne", 1566, 3806092, 646.9])
table.add_row(["Perth", 5386, 1554769, 869.4])

table.set_style(TableStyle.SINGLE_BORDER)
table.set_style(TableStyle.DEFAULT)  # <!-- Add this
table.set_style(TableStyle.MARKDOWN)

print(table)
```

Outputs:

```md
| City name | Area | Population | Annual Rainfall |
| :-------: | :--: | :--------: | :-------------: |
|  Adelaide | 1295 |  1158259   |      600.5      |
|  Brisbane | 5905 |  1857594   |      1146.4     |
|   Darwin  | 112  |   120900   |      1714.7     |
|   Hobart  | 1357 |   205556   |      619.5      |
|   Sydney  | 2058 |  4336374   |      1214.8     |
| Melbourne | 1566 |  3806092   |      646.9      |
|   Perth   | 5386 |  1554769   |      869.4      |
```

Renders:

| City name | Area | Population | Annual Rainfall |
| :-------: | :--: | :--------: | :-------------: |
|  Adelaide | 1295 |  1158259   |      600.5      |
|  Brisbane | 5905 |  1857594   |      1146.4     |
|   Darwin  | 112  |   120900   |      1714.7     |
|   Hobart  | 1357 |   205556   |      619.5      |
|   Sydney  | 2058 |  4336374   |      1214.8     |
| Melbourne | 1566 |  3806092   |      646.9      |
|   Perth   | 5386 |  1554769   |      869.4      |

**Comment by hugovk** [comment]
We should probably call `_set_default_style()` at the start of `set_style()` for all styles (and remove `_set_default_style()` from the start of `_set_orgmode_style()`.

Would you like to submit a PR with tests?

**Comment by bmihaila-bd** [comment]
Yes that would make sense to make sure all other styles are clean, too.

**Comment by bmihaila-bd** [comment]
Regarding doing the PR, sadly I don't have the time for that at the moment, sorry. I added the workaround in my code to call `_set_default_style()` and moved on to next things. Can try in a couple of weeks but hard to promise a timeline.

**Comment by hugovk** [comment]
Sure, no rush :)

As a workaround, you might prefer to call `set_style(TableStyle.DEFAULT)` rather than the internal `_set_default_style()`.

**Comment by MonstersInc-sudo** [comment]
@hugovk , I took a look at the code and added some logic for resetting defaultsbefore going to if statement in set_style along with taking out redundant resets in the _set_orgmode method.

Pull request is here. [PR-344](https://github.com/prettytable/prettytable/pull/344). Will upload tests shortly.

**Comment by hugovk** [comment]
Fixed by https://github.com/prettytable/prettytable/pull/344, thanks all!
